docs/program-setup/bare_metal/baremetal_usage.md

```diff
@@ -0,0 +1,75 @@
+# Using bare metal profile
+
+## Creating a bare metal application
+
+First, clone Mbed OS into the directory where we want to create an application
+```
+mkdir my_mbed_app && cd my_mbed_app
+git clone https://github.com/ARMmbed/mbed-os.git
+```
+
+To enable Mbed OS bare metal, create an `mbed_app.json` with the following contents:
+```
+{
+    "requires": ["bare-metal"]
+}
+```
+
+On bare metal, features are enabled with a bottom-up approach - those listed in [Bare metal APIs](bare_metal_api.md#bare-metal-apis) are now ready for use (but not other features) in our application. See [Bare metal example](baremetal_example.md).
+
+To pull in additional features (take filesystem for example)
+
+1. Locate the libraries we want to use in `mbed-os/` and find their names in their respective `mbed_lib.json`, e.g. `features/storage/filesystem/mbed_lib.json`:
+    ```json
+    {
+        "name": "filesystem",
+        "config": {
+            "present": 1
+        }
+    }
+    ```
+1. Add the libraries to `"requires"` in our application's `mbed_app.json`, for example:
+    ```json
+    {
+        "requires": [
+            "bare-metal",
+            "filesystem"
+        ]
+    }
+    ```
+1. Write the application to use bare metal APIs and libraries added in 2.
+1. Try to compile:
+    ```
+    mbed compile -t <TOOLCHAIN> -m <TARGET>
+    ```
+    Compilation may fail if any libaries added in 2 depend on other libraries. Use error messages to locate dependent libraries (like we did in 1.) and add them to `mbed_lib.json` too, for example `"filesystem"` depends on `"fat_chan"`:
+    ```
+    {
+        "requires": [
+            "bare-metal",
+            "filesystem",
+            "fat_chan"
+        ]
+    }
+    ```
+    Repeat this step until compilation succeeds, i.e. we have added all dependencies.
+
+## Running Greentea tests for bare metal profile
+
+(If you have not used Greentea before, [you can learn more in our documentation](../tools/greentea-testing-applications.html).)
+
+To run Greentea tests for bare metal profile, simply use `baremetal.json`. The full commands are as follows:
+
+1. Change directory:
+
+    ```
+    cd mbed-os
+    ```
+
+1. Execute the Greentea test suite with the bare metal configuration for the supported toolchains:
+
+    ```
+    mbed test -m <YOUR_TARGET> -t <TOOLCHAIN> --app-config TESTS/configs/baremetal.json
```

Comment:
Should we use `--app-config` or `--test-config`?
Can this command also be used with a modified mbed_app.json to change library defaults?

Reply:
Let's keep app-config so we don't confuse users with tests configs.